apmcSpringBoot/src/main/java/com/apmc/apmcSpringBoot/controller/AgentController.java

```diff
@@ -0,0 +1,59 @@
+package com.apmc.apmcSpringBoot.controller;
+
+import com.apmc.apmcSpringBoot.Exception.Response;
+import com.apmc.apmcSpringBoot.Exception.ResponseException;
+import com.apmc.apmcSpringBoot.model.Agent;
+import com.apmc.apmcSpringBoot.service.AgentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/agent")
+public class AgentController {
+
+    @Autowired
+    private AgentService agentService;
+
+    @GetMapping("")
+    public ResponseEntity<List<Agent>> getAllAgents(){
+        return ResponseEntity.ok(agentService.getAllAgent());
+    }
+
+    @GetMapping("/{agentId}")
+    public Agent getAgentById(@PathVariable("agentId") int agentId){
+        Agent agent = agentService.getAgentById(agentId);
+        if(agent == null){
+            throw new ResponseException("Not agent found for this Id");
```

Comment:
I think it will make sense to create a customException: DoesNotExistException for such cases